### PR TITLE
Fixing date to display if you're logged in

### PR DESCRIPTION
**Pull request by jkachel**
# What are the relevant tickets?

https://github.com/mitodl/hq/issues/2902

# Description (What does it do?)

If you were logged in but not enrolled in a course, the More Dates link would work but not show anything. This fixes that - we were looking for you to be an anonymous user to display the link. 

# How can this be tested?

Log in and view a course with multiple course runs that you're not enrolled in. Clicking More Dates should show you the additional dates.

### CODE REVIEW

File: frontend/public/src/components/CourseInfoBox.js [hunk 1]
```diff
@@ -51,11 +51,7 @@ export default class CourseInfoBox extends React.PureComponent<CourseInfoBoxProp
   }
 
   renderUnenrolledDateLink(run: EnrollmentFlaggedCourseRun) {
-    const { currentUser } = this.props
-
-    return !currentUser || !currentUser.id ? (
-      <>{getStartDateForRun(run)}</>
-    ) : null
+    return <>{getStartDateForRun(run)}</>
```
Comment (annagav) [inline frontend/public/src/components/CourseInfoBox.js:54]: since this is so simple now, do we still need this function?

**Review by annagav — CHANGES_REQUESTED** [review]
Maybe we could just eliminate `renderUnenrolledDateLink` function all together.  And have it like this:
```
? this.renderEnrolledDateLink(courseRun)
 : getStartDateForRun(run)}
```

**Review by annagav — APPROVED** [review]
👍

**Comment by jkachel** [comment]
@annagav can you give this a quick re-look? functionality should be the same, but have now added tests for the functionality that broke.

**Comment by annagav** [comment]
LGTM 👍

**Pull request merged**